appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroBase.kt

```diff
@@ -139,26 +149,46 @@ abstract class AppIntroBase : AppCompatActivity(), AppIntroViewPager.OnNextPageR
         pagerAdapter.notifyDataSetChanged()
     }
 
+
+    /**
+     * Use this method to associate a permission with a slide.
+     * Please note that the permission will not be mandatory and the user can choose to deny it and move forward.
+     * If you want the permission to be granted before moving forward, use  [.askForPermissions]
+     *
+     * @param permissions
+     * @param slidesNumber
+     */
+    protected fun askForPermissions(permissions: Array<String>, slidesNumber: Int) {
+        askForPermissions(permissions, slidesNumber, false)
+    }
+
     /**
-     * Request for a specific set of permission at a given slide.
-     * @param permissions A list of permission Strings
-     * @param slideNumber The slide position where to ask for permission
+     * Called by the user to associate permissions with slides.
+     *
+     * @param permissions  - Array of permissions
+     * @param slidesNumber - The slide at which permission is to be asked.
+     * @param required     - Whether the user can change this slide without granting the permissions.
      */
-    protected fun askForPermissions(permissions: Array<String>, slideNumber: Int) {
+    protected fun askForPermissions(permissions: Array<String>, slidesNumber: Int, required: Boolean) {
```

Comment:
Please don't overload this method but provide a default parameter for `required`. Plus annotate with `@JvmOverloads`